From dcb563cd8a0c2507ef088da42421206c81f6466f Mon Sep 17 00:00:00 2001
From: Grigory Kletsko <grigory.kletsko@cogentembedded.com>
Date: Fri, 17 Feb 2017 03:23:48 +0300
Subject: [PATCH] Fix crash when touchscreen generates out of screen coordinate

---
 src/input.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/input.c b/src/input.c
index e230c83..cd4d518 100644
--- a/src/input.c
+++ b/src/input.c
@@ -245,11 +245,14 @@ default_grab_touch_down(struct weston_touch_grab *grab, uint32_t time,
 	struct wl_list *resource_list;
 	wl_fixed_t sx, sy;
 
+	if (!touch->focus)
+		return;
+
 	weston_view_from_global_fixed(touch->focus, x, y, &sx, &sy);
 
 	resource_list = &touch->focus_resource_list;
 
-	if (!wl_list_empty(resource_list) && touch->focus) {
+	if (!wl_list_empty(resource_list)) {
 		serial = wl_display_next_serial(display);
 		wl_resource_for_each(resource, resource_list)
 				wl_touch_send_down(resource, serial, time,
-- 
2.7.4

